### PR TITLE
Notify users that envelope was discarded and retry sending it

### PR DESCRIPTION
**Pull request by dshulyak**
Changes in the whisper that enable this feature: https://github.com/status-im/whisper/pull/23

After some experimentation i decided to extend existing signal. This signal will be extended with a Message keyword. When we receive error from a peer that envelope was discarded because of the time sync issue - Message will be populated with appropriate error.

So, basically from the client perspective this change doesn't introduce any new steps in the flow of signals. Client still waits for an ExpiredSignal. But now we can provide more feedback why certain envelopes weren't delivered.

### CODE REVIEW

File: services/shhext/envelopes.go
Comment (pedropombeiro) [inline services/shhext/envelopes.go:202]: Why not 

```suggestion
	for i, envelopeError := range envelopeErrors {
```

File: signal/events_shhext.go
Comment (pedropombeiro) [inline signal/events_shhext.go]:
```suggestion
	// EventEnvelopeDiscarded is triggered when the envelope was discarded by a peer for some reason.
```

**Review by pedropombeiro — APPROVED** [review]
lgtm, added a couple minor comments

File: services/shhext/envelopes.go
Comment (dshulyak) [reply services/shhext/envelopes.go:202]: to avoid copying object. EnvelopeError is not a pointer, so full object will be copied if accessed this way

Comment (pedropombeiro) [reply services/shhext/envelopes.go:202]: gotcha

**Comment by dshulyak** [comment]
@mandrigin @adambabik can i get a second review here please?

**Review by adambabik — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]

**Review by mandrigin — APPROVED** [review]

**Pull request merged**
